src/web_peer_connection.cpp

```diff
@@ -411,7 +411,16 @@ void web_peer_connection::write_request(peer_request const& r)
 			auto redirection = m_web->redirects.find(f.file_index);
 			if (redirection != m_web->redirects.end())
 			{
-				request += redirection->second;
+				auto const& redirect = redirection->second;
+				if (using_proxy && !redirect.empty() && (redirect[0] == '/'))
```

Comment:
a more concise way of expressing this would be:

```
bool const trailing_slash = using_proxy && !redirect.empty() && redirect[0] == '/';
request.append(redirect, trailing_slash, std::string::npos);
```